pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -157,6 +165,18 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 			} else if shouldPodBeInEndpoints(pod) {
 				if key != pc.podsByIP[ip] {
 					pc.update(ip, key)
+				} else if !model.WorkloadInstancesEqual(pc.c.workloadInstancesByPods[key], instance) {
+					// fire off eds update on pod change
+					svcList, err := pc.c.serviceLister.Services(pod.Namespace).List(k8sLabels.Everything())
+					if err != nil {
+						return err
+					}
+					for _, svc := range svcList {
+						// look for matched services and update eds
+						if labels.Instance(svc.Spec.Selector).SubsetOf(pod.Labels) {
```

Comment:
there may be performance concerns about O(n) lookup on services for every pod change

Reply:
Yes, but I didn't figure out a better solution, the same pattern can be found in workload instance handlers

https://github.com/istio/istio/blob/bc83862fbc62ca83493545ab332bdd68fe4b7e6b/pilot/pkg/serviceregistry/kube/controller/controller.go#L1168-L1177
https://github.com/istio/istio/blob/bc83862fbc62ca83493545ab332bdd68fe4b7e6b/pilot/pkg/serviceregistry/serviceentry/servicediscovery.go#L441-L445